### PR TITLE
Skip registering listener in observermode

### PR DESCRIPTION
**Pull request by beylmk**
**this one is for merging into main, for our RC2**

First step of our workaround for Amazon observer mode. Avoid registering a listener when in observerMode, because Amazon does not support registering two listeners, and we want the user's listener to be triggered.

Next step is to provide another method for syncingPurchases in Amazon observer mode...meaning the error log here is temporary.

[sc-12917]

### CODE REVIEW

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt [hunk 1]
```diff
@@ -41,6 +41,7 @@ internal class AmazonBilling constructor(
     private val applicationContext: Context,
     private val amazonBackend: AmazonBackend,
     private val cache: AmazonCache,
+    private val observerMode: Boolean = false,
```
Comment (vegaro) [inline feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt:44]: Is this default needed?

File: purchases/src/main/kotlin/com/revenuecat/purchases/BillingFactory.kt [hunk 2]
```diff
@@ -28,7 +29,7 @@ object BillingFactory {
             try {
                 Class.forName("com.revenuecat.purchases.amazon.AmazonBilling")
                     .getConstructor(Context::class.java, Backend::class.java, DeviceCache::class.java)
```
Comment (vegaro) [inline purchases/src/main/kotlin/com/revenuecat/purchases/BillingFactory.kt:31]: The constructor has changed, so this constructor doesn't exist anymore and this line needs to be updated. It is not ideal this compiles but we are instantiating using reflection 😢

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt [hunk 3]
```diff
@@ -152,6 +157,8 @@ internal class AmazonBilling constructor(
         replaceSkuInfo: ReplaceSkuInfo?,
         presentedOfferingIdentifier: String?
     ) {
+        if (checkObserverMode()) return
```
Comment (beylmk) [inline feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt:160]: i added this check before any interaction with the purchasingservice...based on [docs](https://developer.amazon.com/docs/in-app-purchasing/iap-implement-iap.html#purchasingservice), none of those calls will work without the listener registered

File: feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt [hunk 4]
```diff
@@ -352,4 +364,12 @@ internal class AmazonBilling constructor(
     private fun onPurchaseError(error: PurchasesError) {
         purchasesUpdatedListener?.onPurchasesFailedToUpdate(error)
     }
+
+    private fun checkObserverMode(): Boolean {
+        return if (observerMode) {
+            log(LogIntent.AMAZON_ERROR, AmazonStrings.ERROR_OBSERVER_MODE_NOT_SUPPORTED)
```
Comment (beylmk) [inline feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt:370]: wondering if this should be a warning, since we may call some of the unsupported methods ourselves...

**Comment by beylmk** [comment]
@vegaro ready for re-review

**Review by vegaro — APPROVED** [review]

File: examples/MagicWeather/app/build.gradle [hunk 5]
```diff
@@ -48,8 +48,8 @@ android {
 }
 
 dependencies {
-    implementation 'com.revenuecat.purchases:purchases:$purchases_version'
-    implementation 'com.revenuecat.purchases:purchases-store-amazon:$purchases_version'
+    implementation "com.revenuecat.purchases:purchases:$purchases_version"
+    implementation "com.revenuecat.purchases:purchases-store-amazon:$purchases_version"
```
Comment (vegaro) [inline examples/MagicWeather/app/build.gradle:52]: Sorry to be picky but do you mind extracting these unrelated changes in MagicWeather to another branch? That way it's easier to revert and cherry pick changes :)

Comment (beylmk) [reply examples/MagicWeather/app/build.gradle:52]: yep no problem my bad!
